components/infobox/commons/infobox_person.lua

```diff
@@ -0,0 +1,280 @@
+	end
+
+	return infobox:build()
+end
+
+--- Allows for overriding this functionality
+function Player:getStorageType(args, role, status)
+	return 'player'
+end
+
+--- Allows for overriding this functionality
+function Player:getInformationType(args)
+	return args.informationType or 'Player'
+end
+
+--- Allows for overriding this functionality
+function Player:adjustLPDB(lpdbData, args, role, status)
+	return lpdbData
+end
+
+--- Allows for overriding this functionality
+function Player:getRole(args)
+	return { display = args.role, store = args.role, category = args.role or 'Player'}
+end
+
+--- Allows for overriding this functionality
+function Player:getStatus(args)
+	return { display = args.status, store = args.status }
+end
+
+--- Allows for overriding this functionality
+function Player:getHistory(infobox, args)
+	return args.history
+end
+
+--- Allows for overriding this functionality
+--- Decides if we store in LPDB and Vars or not
+function Player:shouldStoreData(args)
+	return Namespace.isMain()
+end
+
+--- Allows for overriding this functionality
+--- e.g. to add faction icons to the display for SC2, SC, WC
+function Player:nameDisplay(args)
+	local team = args.teamlink or args.team
+	local icon = mw.ext.TeamTemplate.teamexists(team)
+		and mw.ext.TeamTemplate.teamicon(team) or ''
+	local name = args.id or mw.title.getCurrentTitle().text
+
+	return icon .. '&nbsp;' .. name
+end
+
+--- Allows for overriding this functionality
+function Player:getAchievements(infobox, args)
+	return args.achievements
+end
+
+--- Allows for overriding this functionality
+function Player:calculateEarnings(args)
+	return 0
+end
+
+function Player:_createRegion(region)
+	if region == nil or region == '' then
+		return ''
+	end
+
+	return Template.safeExpand(self.frame, 'Region', {region})
+end
+
+function Player:_createLocations(args, role)
+	local countryDisplayData = {}
+	local country = args.country or args.country1 or args.nationality or args.nationality1
+	if country == nil or country == '' then
+		return countryDisplayData
+	end
+
+	countryDisplayData[1] = Player:_createLocation(country, args.location, role)
+
+	for index = 2, 99 do
+		country = args['country' .. index] or args['nationality' .. index]
+		if country == nil or country == '' then
+			break
+		else
+			countryDisplayData[index] = Player:_createLocation(country, args['location' .. index], role)
+		end
+	end
+
+	return countryDisplayData
+end
+
+function Player:_createLocation(country, location, role)
+	if country == nil or country == '' then
+		return nil
+	end
+	local countryDisplay = Flags._CountryName(country)
+	local demonym = Localisation(countryDisplay)
+
+	return Flags._Flag(country) .. '&nbsp;' ..
+				'[[:Category:' .. countryDisplay .. '|' .. countryDisplay .. ']]'
+				.. '[[Category:' .. demonym .. ' ' .. role .. ']]'
+				.. (location ~= nil and (',&nbsp;' .. location) or '')
+end
+
+function Player:_createTeam(team, link)
```

Comment:
Resolve potential redirect?

Reply:
Just had a thought, isn't it better to do mw.ext.TeamTemplate.team(team) here? 
Then we can deprecate the link parameters & the displayname of team name is their standardized tt name instead of whatever the contributor entered in wikicode.

Reply:
it is only for display, why would we need to resolve redirect for display?

Reply:
TT with a fallback on Template:Team and/or manual?

Reply:
issue is that on some wikis the pagenames do not match the tt names
so if there is no redirect it will fail, so we need some fall backs for sure
will work on it after lunch^^

Reply:
TT.team() takes the makes a link that looks like this (simplify psuedo)
`[File:(TT.raw().icon or TT.raw().legacyimage)] [[TT.raw().pagename | TT.raw().teamname]`
so the teampage doesn't need to match the tt name, only the tt pagename need setup with the correct page. And If the tt pagename isn't setup correctly, it will fail for this team everywhere else, like on the match list on the front page, in the brackets, etc.